common/cache/domainCache.go

```diff
@@ -818,37 +825,69 @@ func (entry *DomainCacheEntry) GetFailoverEndTime() *int64 {
 	return entry.failoverEndTime
 }
 
-// IsActive return whether the domain is active, i.e. non global domain or global domain which active cluster is the current cluster
-// If domain is not active, it also returns an error
-func (entry *DomainCacheEntry) IsActiveIn(currentCluster string) (bool, error) {
+// NewDomainNotActiveError return a domain not active error
+// currentCluster is the current cluster
+// activeCluster is the active cluster which is either domain's active cluster or it's inferred from workflow task version
+func (entry *DomainCacheEntry) NewDomainNotActiveError(currentCluster, activeCluster string) *types.DomainNotActiveError {
+	if entry.GetReplicationConfig().IsActiveActive() {
+		return &types.DomainNotActiveError{
+			Message: fmt.Sprintf(
+				"Domain: %s is active in cluster(s): %v, while current cluster %s is a standby cluster. Operation active cluster: %s",
+				entry.GetInfo().Name,
+				entry.activeClusters,
+				currentCluster,
+				activeCluster,
+			),
+			DomainName:     entry.GetInfo().Name,
+			CurrentCluster: currentCluster,
+			ActiveCluster:  activeCluster,
+			ActiveClusters: entry.activeClusters,
```

Comment:
when do we use this field?

Reply:
This is informational only for error is inspected. No behavior attached to it. I considered making use of it in forwarding layer but not in this initial iteration. We need to attach more info here such as last_update_time of domains to compare with current cluster's domain config to decide which one to respect etc.